src/python/pants/goal/run_tracker.py

```diff
@@ -408,19 +410,24 @@ def run_information(self):
       run_information['target_data'] = ast.literal_eval(target_data)
     return run_information
 
+  def _stats(self):
+    if self.get_options().stats_version == 1:
+      return {
+        'run_info': self.run_information(),
+        'cumulative_timings': self.cumulative_timings.get_all(),
+        'self_timings': self.self_timings.get_all(),
+        'critical_path_timings': self.get_critical_path_timings().get_all(),
+        'artifact_cache_stats': self.artifact_cache_stats.get_all(),
+        'pantsd_stats': self.pantsd_stats.get_all(),
+        'outcomes': self.outcomes,
+        'recorded_options': self._get_options_to_record(),
+      }
+    else:
+      return json.loads(open(self.report.json_reporter.report_path).read())
```

Comment:
AFAICT there is no local guaranty self.report.json_reporter won't raise `KeyError`. It looks like this is arranged here:
https://github.com/pantsbuild/pants/blob/76415e5902672b70965568acf42f12eca2a32b1b/src/python/pants/reporting/reporting.py#L84-L109

It would be nice though to be able to read just the files in this PR and know everything is going to be OK.